core/server/src/main/java/alluxio/master/file/options/CreatePathOptions.java

```diff
@@ -39,6 +40,7 @@ protected CreatePathOptions() {
     mPersisted = false;
     mRecursive = false;
     mMetadataLoad = false;
+    mDefaultMode = true;
```

Comment:
it just feels like we should not be using Permission as a field and instead there should be three different fields (owner, group, mode) ... I feel like we had this conversation before

public API or not, your change introduces relationship between two fields of the CreatePathOptions class and the correct use of the class now requires coordinated invocations of getPermission + isDefaultMode or setPermission + setDefaultMode

1. please add a TODO to CreatePathOptions to replace Permission with three fields (if you think it is a good idea)

2. change getPermission() and setMode() in CreateFileOptions as follows:

```
public Permission getPermission() {
  Permission permission = new Permission(mPermission);
  if (mDefaultMode) {
    permission.setMode((Mode.getDefault()).applyFileUMask());
  }
  return permission;
}

public CreateFileOptions setMode(Mode mode) {
  mMode = mode;
  mDefaultMode = false;
}
```

and similarly for CreateDirectoryOption. That way, the only place where we have to explicitly modify DefaultMode from outside of the class is when create the options for the recursive directory creation.